### PR TITLE
Grouped operations (and more)

### PR DESCRIPTION
**Pull request by valpesendorfer**
This PR came from improvement work done for pre-calculating SPIs and such for admin extractions.

Most notably this PR adds:

- computation of groups along time dimension
    - ` gammastd_grp` for grouped SPIs, that can be calculated by passing an array with group indices to the `.algo.spi` accessor
    - `mean_grp` grouped means (accessible through `algo.mean_grp` accessor)
- a new accessor `rolling` that allows for rolling computations over a defined window. Currently, `rolling.sum` is implemented
- a fix for `zonal.mean` which prevented the definition of output datatype

### CODE REVIEW

**Review by Kirill888 — APPROVED** [review]

File: hdc/algo/ops/zonal.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 @lazycompile(njit)
-def do_mean(pixels, z_pixels, num_zones, nodata, z_nodata, dtype="float64"):
+def do_mean(pixels, z_pixels, num_zones, nodata, z_nodata, out_dtype=np.float32):
```
Comment (Kirill888) [inline hdc/algo/ops/zonal.py:9]: why this change is necessary? don't see any matching call-site changes

Comment (Kirill888) [reply hdc/algo/ops/zonal.py:9]: ok, I see why. Usually this problem of passing extra args to dask graph is best done with `functools.partial(my_func, **fixed_named_args_that_might_alias)`

**Review by Kirill888 — APPROVED** [review]

File: hdc/algo/ops/zonal.py (hunk 1 above)
Comment (valpesendorfer) [reply hdc/algo/ops/zonal.py:9]: There were a couple of problems with the previous implementation:

- `dtype` is already a keyword to `map_blocks`, so the naming is ambiguous when passed in through the accessor (which could be done with `functools.partial` I suppose.
- second problem was numba: when passing in a dtype with string notation it would fail compiling. Not sure if that would have been solved by using `functools.partial`, but passing the object works fine

**Comment by valpesendorfer** [comment]
OK this is now ready to be merged.

Quick summary:

## Additions
- computation of groups along time dimension
    - `gammastd_grp` for grouped SPIs, that can be calculated by passing an array with group indices to the `.algo.spi` accessor
    - `mean_grp` grouped means (accessible through `algo.mean_grp` accessor) 
- a new accessor rolling that allows for rolling computations over a defined window. Currently, `rolling.sum` is implemented

## Improvements
- better handling of nodata in SPI methods
- a fix for `zonal.mean`which prevented the definition of output datatype

## Unrelated changes
- surface `Dekad.ndays` from `hdc.algo.dekad.Dekad` to the `Period` accessor of `hdc.algo`

File: hdc/algo/ops/stats.py [hunk 2]
```diff
@@ -217,9 +218,15 @@ def gammastd_yxt(
...
         cal_stop = t
 
+    if nodata is None:
+        nodata = -9999
+
     for ri in range(r):
         for ci in range(c):
             xt = x[ri, ci, :]
+            if (xt != nodata).sum() == 0:
+                y[ri, ci, :] = -9999
```
Comment (Kirill888) [inline hdc/algo/ops/stats.py:228]: `= nodata` ?

Comment (Kirill888) [reply hdc/algo/ops/stats.py:228]: there are still a lot of -9999 constants around

Comment (valpesendorfer) [reply hdc/algo/ops/stats.py:228]: `gammastd` returns `-9999` as fill value. The idea is that `nodata` is used to determine input nodata, and output is fixed to `-9999`

Comment (Kirill888) [reply hdc/algo/ops/stats.py:228]: that's a confusing api then,  usually in/out nodata is the same or there are two options for src/dst. At the very least output nodata value should then be placed into a variable with reasonable name.

**Review by Kirill888 — APPROVED** [review]

**Comment by valpesendorfer** [comment]
Updated summary:

## Additions
- computation of groups along time dimension
    - `gammastd_grp` for grouped SPIs, that can be calculated by passing an array with group indices to the `.algo.spi` accessor
    - `mean_grp` grouped means (accessible through `algo.mean_grp` accessor) 
- a new accessor rolling that allows for rolling computations over a defined window. Currently, `rolling.sum` is implemented

## Improvements
- better handling of nodata in SPI methods
- a fix for `zonal.mean`which prevented the definition of output datatype

## Unrelated changes
- surface from `hdc.algo.dekad.Dekad` to the `Period` accessor of `hdc.algo`:
    -  `Dekad.ndays`
    -  `Dekad.start_date`
    -  `Dekad.end_date`
    -  `Dekad.raw`
